test/lib/ansible_test/_data/sanity/import/importer.py

```diff
@@ -196,24 +213,33 @@ def __load_module(self, fullname):
             self.loaded_modules.add(fullname)
             return import_module(fullname)
 
-    def run():
+    def run(args):
         """Main program function."""
         base_dir = os.getcwd()
         messages = set()
 
-        for path in sys.argv[1:] or sys.stdin.read().splitlines():
-            name = convert_relative_path_to_name(path)
-            test_python_module(path, name, base_dir, messages)
+        if import_type == 'module':
+            for path in args or sys.stdin.read().splitlines():
+                name = convert_relative_path_to_name(path)
+                test_python_module(path, name, base_dir, messages, True)
+        elif import_type == 'plugin':
+            for path in args or sys.stdin.read().splitlines():
+                name = convert_relative_path_to_name(path)
+                test_python_module(path, name, base_dir, messages, False)
+        else:
+            print('Invalid import type!')
+            sys.exit(1)
```

Comment:
This can be simplified:

```suggestion
        for path in args or sys.stdin.read().splitlines():
            name = convert_relative_path_to_name(path)
            test_python_module(path, name, base_dir, messages, restrict_to_module_paths)
```

The extra error logic shouldn't be needed since this is only called internally by ansible-test.